.travis.yml

```diff
@@ -74,136 +90,186 @@ matrix:
         # -> Have package dependency that matches partially another package
         #    (pytest vs pytest-cov) when falling back to pip install
         # -> testing whether version numbers are respected by pip install fallback
-        - os: linux
-          env: SETUP_CMD='test --coverage' PYTHON_VERSION=3.4
-               TEST_CMD='py.test test_env.py && coverage run test_env.py; python -c "import sklearn; assert sklearn.__version__.startswith(\"0.19\")"'
-               CONDA_CHANNELS='astrofrog conda-forge'
-               CONDA_DEPENDENCIES='scipy pyqt5 openjpeg photutils pytest pytest-cov scikit-learn'
-               DEBUG=True NUMPY_VERSION=1.10 ASTROPY_VERSION=stable
-               SCIKIT_LEARN_VERSION='<0.20,>=0.19'
-
+        #
+        # The list of dependencies was considerably simplified to speed up the
+        # (expected) conda failure.
+        #
+        # Need to omit conda-forge from channels here because ipyevents is on conda-forge
+        # and we want the pip fallback to happen.
+        - os: linux
+          env: SETUP_CMD='test --coverage' PYTHON_VERSION=3.6
+               TEST_CMD='conda list; py.test test_env.py && coverage run test_env.py; python -c "import ipyevents; assert ipyevents.__version__.startswith(\"0.4\")"'
+               CONDA_CHANNELS=''
+               CONDA_DEPENDENCIES='libtiff pytest pytest-cov ipyevents'
+               DEBUG=True ASTROPY_VERSION=stable
+               IPYEVENTS_VERSION='<0.5,>=0.4'
+
+        # JOB .8
         # -> The pre builds should run the testing phase only when a pre
         #    release is available on pypi, otherwise it should pass
         - os: linux
           env: SETUP_CMD='test' NUMPY_VERSION=pre
+        # JOB .9
         - os: linux
           env: SETUP_CMD='test' ASTROPY_VERSION=pre DEBUG=True
+        # JOB .10
         - os: linux
           env: SUNPY_VERSION=prerelease CONDA_CHANNELS='conda-forge'
 
+        # JOB .11
         # -> Setting pytest version should be recognized in the next two jobs
-        - os: linux
-          env: SETUP_CMD='build_sphinx' CONDA_DEPENDENCIES='matplotlib<=1.5.0'
-               DEBUG=True PYTHON_VERSION=2.7 PYTEST_VERSION=2.7.3
-               TEST_CMD='python -c "import matplotlib; assert matplotlib.__version__==\"1.5.0\"; import pytest; assert pytest.__version__.startswith(\"2.7.3\")"'
-
-        - os: linux
-          env: SETUP_CMD='build_sphinx' SPHINX_VERSION=1.4.1 DEBUG=True
-               MATPLOTLIB_VERSION=1.4 DEBUG=True PYTHON_VERSION=3.5 PYTEST_VERSION=3.0
-               TEST_CMD='python -c "import sphinx; assert sphinx.__version__==\"1.4.1\"; import pytest; pytest.__version__.startswith(\"3.0\"); import matplotlib; assert matplotlib.__version__=="\""1.4.3"\"'
-
+        #
+        # Changed version numbers in 2019 because older versions are no longer
+        # available in default conda channels:
+        #
+        #    matplotlib 1.5.0 → 2.2.0
```

Comment:
no need for comments in the file, having the updates and these info in the commit message is enough